docs/icicle/primitives/poseidon.md

```diff
@@ -0,0 +1,228 @@
+# Poseidon
+
+Poseidon hash is a popular hash in the ZK ecosystem primarily because its optimized to work over large prime fields, a common setting for ZK proofs, thereby minimizing the number of multiplicative operations required.
```

Comment:
Actually, I thought that the main feature is it's ability to easily fit inside ZK circuits. So computing Poseidon `inside` a ZK circuit is much much faster than Keccak or Sha-256. Maybe I am wrong

Reply:
its also that.

Reply:
Wasnt sure how many to add, but ill add ur point as well.